xmodule/contentstore/mongo.py

```diff
@@ -310,71 +310,47 @@ def _get_all_content_for_course(self,
             contentType: The mimetype string of the asset
             md5: An md5 hash of the asset content
         '''
-        # TODO: Using an aggregate() instead of a find() here is a hack to get around the fact that Mongo 3.2 does not
-        # support sorting case-insensitively.
-        # If a sort on displayname is requested, the aggregation pipeline creates a new field:
-        # `insensitive_displayname`, a lowercase version of `displayname` that is sorted on instead.
-        # Mongo 3.4 does not require this hack. When upgraded, change this aggregation back to a find and specifiy
-        # a collation based on user's language locale instead.
-        # See: https://openedx.atlassian.net/browse/EDUCATOR-2221
-        pipeline_stages = []
         query = query_for_course(course_key, 'asset' if not get_thumbnails else 'thumbnail')
         if filter_params:
             query.update(filter_params)
-        pipeline_stages.append({'$match': query})
 
+        # Count total matching documents
+        count = self.fs_files.count_documents(query)
+
+        sort_list = []
         if sort:
             sort = dict(sort)
             if 'displayname' in sort:
-                pipeline_stages.append({
-                    '$project': {
-                        'contentType': 1,
-                        'locked': 1,
-                        'chunkSize': 1,
-                        'content_son': 1,
-                        'displayname': 1,
-                        'filename': 1,
-                        'length': 1,
-                        'import_path': 1,
-                        'uploadDate': 1,
-                        'thumbnail_location': 1,
-                        'md5': 1,
-                        'insensitive_displayname': {
-                            '$toLower': '$displayname'
-                        }
-                    }
+                # Apply case-insensitive sorting
+                cursor = self.fs_files.find(query, {
+                    'contentType': 1,
+                    'locked': 1,
+                    'chunkSize': 1,
+                    'content_son': 1,
+                    'displayname': 1,
+                    'filename': 1,
+                    'length': 1,
+                    'import_path': 1,
+                    'uploadDate': 1,
+                    'thumbnail_location': 1,
+                    'md5': 1
                 })
-                sort = {'insensitive_displayname': sort['displayname']}
-            pipeline_stages.append({'$sort': sort})
+                cursor = cursor.sort('displayname', sort['displayname']).collation({'locale': 'en', 'strength': 2})
```

Comment:
Suggestion: Definitely not a blocker for this PR, but if we're sending the locale in, we might want to get it all the way from the frontend, so that people using languages other than English can see the sorting they'd expect.